### ISSUE TITLE
azcopy don't work in alpine since 10.25.0

### ISSUE THREAD

**Issue by rnt** [Azure/azure-storage-azcopy#2750]
Since 10.25.0 the azcopy linux_amd64 binary don't works in alpine.

### Which version of the AzCopy was used? 

10.25.0 and 10.25.1

### Which platform are you using? (ex: Windows, Mac, Linux)

Alpine Linux

### What command did you run?

azcopy --version

### What problem was encountered?

Any execution fail with this message:

```
0.104 Error relocating /usr/local/bin/azcopy: __vfprintf_chk: symbol not found
0.104 Error relocating /usr/local/bin/azcopy: __fprintf_chk: symbol not found
```

In 10.24.0, 10.25.0 and 10.25.1, `ldd` output was the same:

```
/ # ldd /usr/local/bin/azcopy
        /lib64/ld-linux-x86-64.so.2 (0x7efd860d0000)
        libpthread.so.0 => /lib64/ld-linux-x86-64.so.2 (0x7efd860d0000)
        libc.so.6 => /lib64/ld-linux-x86-64.so.2 (0x7efd860d0000)
```

### How can we reproduce the problem in the simplest way?

Using the following Dockerfiles:

:green_circle: This build works fine:

```
 docker build . -f Dockerfile.10.24.0.txt -t local-azcopy:10.24.0
```

1. [Dockerfile.10.24.0.txt](https://github.com/user-attachments/files/16270625/Dockerfile.10.24.0.txt)
   ```Dockerfile
   FROM docker.io/library/alpine:3.17
   
   RUN apk update
   RUN apk add --no-cache curl libc6-compat
   RUN curl -s -L https://azcopyvnext.azureedge.net/releases/release-10.24.0-20240326/azcopy_linux_amd64_10.24.0.tar.gz | tar xvzf - --strip-components=1 -C /usr/local/bin/ azcopy_linux_amd64_10.24.0/azcopy
   RUN chown root:root /usr/local/bin/azcopy
   RUN chmod +x /usr/local/bin/azcopy
   RUN ldd /usr/local/bin/azcopy
   RUN azcopy --version
   ```

:red_circle:  This builds don't works:
```
 docker build . -f Dockerfile.10.25.0.txt -t local-azcopy:10.25.0
 docker build . -f Dockerfile.10.25.1.txt -t local-azcopy:10.25.1
```

2. [Dockerfile.10.25.0.txt](https://github.com/user-attachments/files/16270627/Dockerfile.10.25.0.txt)
   ```Dockerfile
   FROM docker.io/library/alpine:3.17
   
   RUN apk update
   RUN apk add --no-cache curl libc6-compat
   RUN curl -s -L https://azcopyvnext.azureedge.net/releases/release-10.25.0-20240522/azcopy_linux_amd64_10.25.0.tar.gz | tar xvzf - --strip-components=1 -C /usr/local/bin/
   RUN chown root:root /usr/local/bin/azcopy
   RUN chmod +x /usr/local/bin/azcopy
   RUN ldd /usr/local/bin/azcopy
   RUN azcopy --version
   ```
3. [Dockerfile.10.25.1.txt](https://github.com/user-attachments/files/16270626/Dockerfile.10.25.1.txt)
   ```Dockerfile
   FROM docker.io/library/alpine:3.17

   RUN apk update
   RUN apk add --no-cache curl libc6-compat
   RUN curl -s -L https://azcopyvnext.azureedge.net/releases/release-10.25.1-20240612/azcopy_linux_amd64_10.25.1.tar.gz | tar xvzf - --strip-components=1 -C /usr/local/bin/
   RUN chown root:root /usr/local/bin/azcopy
   RUN chmod +x /usr/local/bin/azcopy
   RUN ldd /usr/local/bin/azcopy
   RUN azcopy --version
   ```

### Have you found a mitigation/solution?

No. :cry:

**Comment by ashruti-msft** [comment]
You will have to install the dependency that is missing then it should work fine.

**Comment by gapra-msft** [comment]
Closing this issue as a suggestion has been made and there has been no further activity on this thread. Please reopen if you are still experiencing the same issue.

**Comment by niekcandaele** [comment]
For anyone who lands here from google, I had the same issue and was able to fix it. 

I was using the image `docker:24.0.5` based on Alpine 3.18. If you install only `libc6-compat` here, running azcopy fails. If you additionally install `gcompat` -> the binary works.

If you run the same test on a newer Alpine (`docker:28.1.1` in my case, based on Alpine 3.21), when installing `libc6-compat`, it automatically also includes `gcompat`

So TLDR: update your docker image ;)
